### PR TITLE
Check format extension function in ImplicitDefaultLocale

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/6342

### CODE REVIEW

**Comment by atulgpt** [comment]
Hi, @3flex one strange observation here, when I run `./gradlew detektMain` build successfully passes(build scan at https://scans.gradle.com/s/ci2o5a2sfqvkq). But when I run `./gradlew detektMain --offline --rerun-tasks` build fails with below error(build scan at https://scans.gradle.com/s/yilo2rorrwnqi)
```
> Task :detekt-report-html:detektMain FAILED
/Users/atulgup/Learning/open_source/detekt/detekt-report-html/src/main/kotlin/io/github/detekt/report/html/HtmlUtils.kt:31:47: "%1$4s ".format(currentLineNumber) uses implicitly default locale for string formatting. [ImplicitDefaultLocale]
```
I have changed the rule and as per the updated rule it should detect the above issue even ruining `./gradlew detektMain`

File: detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocaleSpec.kt [hunk 1]
```diff
@@ -32,6 +32,52 @@ class ImplicitDefaultLocaleSpec(private val env: KotlinCoreEnvironment) {
         assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
     }
 
+    @Test
+    fun `does not report custom String_format call`() {
+        val code = """
+            fun String.Companion.format(format: String, value: Int) = format + value.toString()
+
+            fun x() {
+                String.format("%d", 1)
```
Comment (BraisGabin) [inline detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocaleSpec.kt:41]: There is a missing test here, right? You aren't testing that this without the extension function should be reported.

Comment (atulgpt) [reply detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/ImplicitDefaultLocaleSpec.kt:41]: Are you talking bout existing `reports String_format call with template but without explicit locale`? This TC is already there. In this TC I am checking our own defined custom method should not report the violation. Earlier we used name matching for detecting the violations and I moved that to `FQName` hence I added this TC

**Review by marschwar — COMMENTED** [review]
Should we also add those new cases to the compliant/non-compliant examples?

**Comment by atulgpt** [comment]
> Should we also add those new cases to the compliant/non-compliant examples?

Updated the code to include the extension method examples

**Review by marschwar — APPROVED** [review]

**Pull request merged**